onnxruntime/core/providers/rocm/rocm_provider_factory.cc

```diff
@@ -128,9 +126,26 @@ struct ProviderInfo_ROCM_Impl : ProviderInfo_ROCM {
   }
 
   // Used by slice_concatenate_test.cc and onnxruntime_pybind_state.cc
-  void rocmMemcpy_HostToDevice(void* dst, const void* src, size_t count) override { HIP_CALL_THROW(hipMemcpy(dst, src, count, hipMemcpyHostToDevice)); }
+
+  void rocmMemcpy_HostToDevice(void* dst, const void* src, size_t count) override {
+    // hipMemcpy() operates on the default stream
+    HIP_CALL_THROW(hipMemcpy(dst, src, count, hipMemcpyHostToDevice));
+
+    // To ensure that the copy has completed, invoke a stream sync for the default stream.
+    // https://docs.nvidia.com/rocm/rocm-runtime-api/api-sync-behavior.html#api-sync-behavior__memcpy-sync
```

Comment:
I don't think this is valid link ;)

Reply:
Fixed in 06c097ea6431c7b56dea43e816b52066dc725e6b.